src/v2/view-builder/views/device/DeviceEnrollmentTerminalView.js

```diff
@@ -1,50 +1,125 @@
-  title() {
-    return loc('enroll.title.mdm', 'login');
+const AndroidAppLinkWithAccountOdaTerminalForm = BaseDeviceEnrollTerminalForm.extend({
+  attributes: { 'data-se': 'android-oda-app-link-with-ov-account-terminal' },
+  title: loc('enroll.title.oda.with.account', 'login'),
+  initialize() {
+    BaseDeviceEnrollTerminalForm.prototype.initialize.apply(this, arguments);
+    this.add(AndroidAppLinkWithAccountOdaTerminalView);
   },
+});
+
+const AndroidAppLinkWithoutAccountOdaTerminalForm = BaseDeviceEnrollTerminalForm.extend({
+  attributes: { 'data-se': 'android-oda-app-link-without-ov-account-terminal' },
+  title: loc('enroll.title.oda.without.account', 'login'),
   initialize() {
-    BaseForm.prototype.initialize.apply(this, arguments);
+    BaseDeviceEnrollTerminalForm.prototype.initialize.apply(this, arguments);
+    this.add(AndroidAppLinkWithoutAccountOdaTerminalView);
+  },
+});
+
+const IosAndAndroidLoopbackOdaTerminalForm = BaseDeviceEnrollTerminalForm.extend({
+  attributes: { 'data-se': 'loopback-terminal' },
+  title: loc('enroll.title.oda', 'login'),
+  initialize() {
+    BaseDeviceEnrollTerminalForm.prototype.initialize.apply(this, arguments);
+    this.add(IosAndAndroidLoopbackOdaTerminalView);
+  },
+});
+
+const MdmTerminalForm = BaseDeviceEnrollTerminalForm.extend({
+  attributes: { 'data-se': 'mdm-terminal' },
+  title: loc('enroll.title.mdm', 'login'),
+  initialize() {
+    BaseDeviceEnrollTerminalForm.prototype.initialize.apply(this, arguments);
     this.add(MdmOktaVerifyTerminalView);
   },
 });
 
+const AndroidAppLinkTerminalViewFooter = Link.extend({
+  postRender() {
+    this.$el.click((event) => {
+      event.preventDefault();
+      this.options.appState.trigger('switchBackToPreselect');
+    });
+  }
+});
+
 export default BaseView.extend({
   initialize() {
     BaseView.prototype.initialize.apply(this, arguments);
+    this.listenTo(this.options.appState, 'updateDeviceEnrollmentView', this.handleUpdateDeviceEnrollmentView);
+    this.listenTo(this.options.appState, 'switchBackToPreselect', this.handleSwitchBackToPreselect);
 
-    const deviceEnrollment = this.options.appState.get('deviceEnrollment');
-    this.enrollmentType = (deviceEnrollment.name || '').toLowerCase(); // oda/mdm
+    const deviceEnrollmentContext = getDeviceEnrollmentContext(this.options.appState.get('deviceEnrollment'));
+    this.enrollmentType = (deviceEnrollmentContext.enrollmentType|| '').toLowerCase(); // oda/mdm
 
     switch (this.enrollmentType) {
     case Enums.ODA:
       this.Header = OktaVerifyAuthenticatorHeader;
-      this.Body = OdaTerminalForm;
+      this.Body = deviceEnrollmentContext.isAndroidAppLink ?
+        AndroidAppLinkPreselectForm : IosAndAndroidLoopbackOdaTerminalForm;
       break;
     case Enums.MDM:
       this.Body = MdmTerminalForm;
       break;
     }
-  }
+  },
+
+  handleUpdateDeviceEnrollmentView(withAccount = 'not-indicated') {
```

Comment:
just wonder what will happen if withAccount is set to default value (which shouldn't happen)?